src/types/annotator.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Type definitions for objects passes between the annotator and sidebar
+ */
+
+/** @typedef {import("./api").Target} Target */
+
+/**
+ * @typedef AnnotationData
+ * @prop {string} uri
+ * @prop {string} $tag
+ * @prop {string} [$highlight]
+ * @prop {Target[]} target
+ * @prop {Document} document
+ */
+
+/**
+ * @typedef Document
```

Comment:
Can I suggest renaming this to say, DocumentMetadata` to avoid confusion with the DOM `Document` interface.

Reply:
indeed, better name.